### DISCUSSION TITLE
BaseTexture color hue, any ideas how to solve this?

### DISCUSSION THREAD

**Discussion by kulabros** [pixijs/pixijs#7569] [Q&A]
Hi there all,

I have a PIXI.Spritesheet made from BaseTexture, and I'd like to change hue of the loaded image, or the existing basetexture, to create PIXI.Spritesheet instance, for the same assets, but with color hue rotation. I'm able to let's say simulate, using rendere's generate texture, but that's just capturing the stage with the original sprite with ColorMatrixFilter. Is there a chance to do this operation with filters in some easy way offscreen? Or do you have any other ideas, how to color hue loaded spritesheet, similar way as imagemagick would do with command -modulate 100,100, hueAngle ? ... There are Spine animations made from the spritesheet images, and it's not possible to tint the images (it's not hue) and setting the filter to the animation container results in badly drawn images with blend mode in the animation... Of course I can serve and load hued image variants, but because of data optimization, I'd like to load the original bitmap and solve the hue in the application. Some tips how to solve this in the PIXI? Or should I source imagemagick to the project... Thank for any tips or help. have a nice day!

**Comment by ivanpopelyshev** [comment]
Our current approach is texture resource. You create texture resource that takes input (in this case: canvas), does something with it and uploads to webgl. How to copy the spritesheet textures - well, that you can do yourself, its not that difficult, clone textures and change their baseTexture i guess. 

Look at https://pixijs.io/examples/#/textures/gradient-resource.js, and try to extend CanvasResource or BaseImageResource (find its sources in this repo)

>  There are Spine animations made from the spritesheet

Oh that's problem. spine loader is very complicated thing, maybe you'll have to hack it. Usually you can pass an existing atlas to it 
1. load texture in one loader, create your CanvasResource based on `data` field of loader resource. 
2. Pass that thing to `metadata.something` field in spine resource according to examples from `pixi-spine` repo.

if that doesnt work for you, you have to debug, a lot. If its not enough, you have to hack spine loader yourself: override spine base loader from `@pixi-spine/loader-base` (not exact name, i dont remember), make your own bundle, copy index.ts from `@pixi-spine/all-3.8`, and voila! you have it.

Yes, hue isn't easy. Yes, spine loader is fubar. Yes, there are not enough examples. However all those things could be worse if some people didnt make PR's to spine to improve that loader through years. And when you find something there that can be improved to help other people - please do it.
